contribute/design/1.x/cstor/20200629-cstor-backupandrestore-v1.md

```diff
@@ -0,0 +1,279 @@
+- CStorRestore			-- (Reconciled by Restore controller in pool manger)
+- CStorCompletedBackup	-- (Used to store last completed and second last completed backup information which is used for scheduled backups)
+
+A detailed description on how cStor Backup and Restore works is out of the scope of this document. For more details to know how the cStor backup and restore works, please click [here](https://github.com/openebs/openebs/blob/master/contribute/design/1.x/cstor/20200511-cstor-backupandrestore.md)
+
+## Motivation
+
+- Promote backup and restore CRs to V1 version of cstor.openebs.io group which provides backup and restore support of cStor CSI volumes.
+
+### Goals
+
+- Freeze the schema of all the CRs related to cStor Backup and Restore.
+- Move the CRs to the cstor.openebs.io  group from the existing openebs.io group.
+- The CRs in  cstor.openebs.io will be in v1 version.
+- Workflow for migration of existing Backup and Restore(i.e. openebs.io group and v1alpha1 API version ) to cstor.openebs.io group and v1 API version should be provided without any downtime and with backward compatibility.
+
+### Non-Goals
+
+- Non-CSI based volumes will not deal with a new version of backup and restore. It will continue with the existing version of backup and restore.
+
+## Proposal
+
+### Proposed Approaches
+
+- Backward compatibility to support any version of backup and restore.
+
+  Example:
+  |   CVC-Operator Version    | Pool-Manager Version  |
+  |  ------------------------ | --------------------  |
+  |   1.11.0                  |  1.11.0               |
+  |   >= 1.12.0               |  >= 1.11.0            |
+
+*Note:*
+
+_CVC-Operator and pool-manager version **< 1.11.0** will not support Backup and Restore_
+
+### Steps to perform user stories
+
+- User need to install or upgrade the CSPC based cStor pools to atleast 1.12.0
+- SPC based cStor pools and Non-CSI volumes should migrate to 1.13.0 version of CSPC-Operater to get v1 version of Backup and Restore.
+
+### Low Level Design
+
+#### Work Flow
+
+##### Current State of Schemas
+
+Following is the existing CStorBackup schema in go struct:
+```go
+// CStorBackup describes a cstor backup resource created as a custom resource
+type CStorBackup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              CStorBackupSpec   `json:"spec"`
+	Status            CStorBackupStatus `json:"status"`
+}
+
+// CStorBackupSpec is the spec for a CStorBackup resource
+type CStorBackupSpec struct {
+	// BackupName is a name of the backup or scheduled backup
+	BackupName string `json:"backupName"`
+
+	// VolumeName is a name of the volume for which this backup is destined
+	VolumeName string `json:"volumeName"`
+
+	// SnapName is a name of the current backup snapshot
+	SnapName string `json:"snapName"`
+
+	// PrevSnapName is the last completed-backup's snapshot name
+	PrevSnapName string `json:"prevSnapName"`
```

Comment:
Existing names on backupSpec(related to comments on PR https://github.com/openebs/api/pull/58#discussion_r459234933)
cc: @vishnuitta

Reply:
Ignore this comment